models/double_entry_transactions/int_quickbooks__bill_payment_double_entry.sql

```diff
@@ -45,7 +58,9 @@ bill_payment_join as (
             order by bill_payments.source_relation, bill_payments.transaction_date) - 1 as index,
         bill_payments.transaction_date,
         bill_payments.total_amount as amount,
-        (bill_payments.total_amount * coalesce(bill_payments.exchange_rate, 1)) as converted_amount,
+        case when bill_payment_lines.journal_entry_id is not null 
+             then ((-1) * bill_payment_lines.amount)* (coalesce(coalesce(bills.exchange_rate, bill_payments.exchange_rate), 1))
+             else (bill_payment_lines.amount * coalesce(coalesce(bills.exchange_rate, bill_payments.exchange_rate), 1)) end as converted_amount,
```

Comment:
Why are we taking the approach of using the `bill_payments.total_amount` for the `amount` field in this model. However, for the `converted_amount` we're using the `bill_payments_line.amount` field. Doesn't this result in a drastic differences between amount and converted_amount? Especially when there are multiple bill payment lines?

For context, when I'm running this with some test data I see that the converted amounts are different between dev and prod, but I'm not sure this is expected. See internal ticket for screenshots.

Reply:
Agreed and moved around to do the line amounts separately, then aggregate them before joining them in.